lib/factory_bot/strategy/stub.rb

```diff
@@ -44,15 +44,17 @@ def next_id
       end
 
       def stub_database_interaction_on_result(result_instance)
-        result_instance.id ||= next_id
+        if has_settable_id?(result_instance)
+          result_instance.id ||= next_id
+        end
 
         result_instance.instance_eval do
           def persisted?
-            !new_record?
+            true
           end
 
           def new_record?
-            id.nil?
+            false
```

Comment:
for completeness maybe we could do the second part of the ```has_settable_id``` method here?

Reply:
Do you mean `result_instance.class.primary_key`? I don't follow how that relates to this method.

Reply:
sorry, I meant the instance method primary key instead of ```id.nil?``` ```primary_key.nil?```

Reply:
`build_stubbed` objects should always act like they are `persisted?`, so I think returning `false` is OK here. Checking whether the primary_key is `nil` still wouldn't work if there is no primary key at all.